### DISCUSSION TITLE
Testing embedded system via QtWebEngine remote debugging

### DISCUSSION THREAD

**Discussion by wappla123** [cypress-io/cypress#27111] [Questions and Help]
Hello everyone,

I´m curious if someone has some experience with this. We are running our application in a minibrowser using the QtWebEngine on an embedded system. We can access and control the instance of the minibrowser through the remote debugging tool provided. I managed to run my cypress tests directly on the hardware but it seems that it is done in another instance. Does anyone know if it is possible to run it in the same instance so you can basically follow the actions on the screen of the HW?

Thx in advance for your time and insights!

**Comment by portyu9** [comment]
Hi @wappla123 

There is no supported Cypress API for attaching a test run to an **already-running QtWebEngine instance** through its remote-debugging port.

Cypress deliberately launches and controls its own browser instance, with its own profile and automation connection. That is why running Cypress on the hardware gives you another browser instance instead of driving the minibrowser that is already displayed on the device.

Cypress can launch supported Chromium-family browser binaries, including custom binary paths, but that is different from connecting to an existing QtWebEngine page/session.

So if the requirement is specifically:

> run the Cypress commands inside the existing QtWebEngine instance that is visible on the embedded device

Cypress is not the right integration point for that requirement. You would need tooling that can directly drive the existing QtWebEngine/Chromium remote-debugging endpoint, or test the same web application separately in a browser launched by Cypress.

There are internal Cypress mechanisms related to browser connections, but they are not a public/supported API for attaching Cypress to an arbitrary existing Chromium/QtWebEngine session.

Please mark this answer as accepted if it helped, thank you.
